node/src/components/consensus/era_supervisor.rs

```diff
@@ -608,11 +688,21 @@ where
         let seed =
             EraSupervisor::<I>::era_seed(booking_block_hash, block_header.accumulated_seed());
         trace!(%seed, "the seed for {}: {}", era_id, seed);
+        let slashed = era_id
+            .iter_other(
+                self.era_supervisor.protocol_config.last_upgrade_point,
+                self.era_supervisor.bonded_eras,
+            )
+            .flat_map(|e_id| &self.era_supervisor.active_eras[&e_id].newly_slashed)
```

Comment:
For the first era we initialize we won't have that. We should take that information out of the switch blocks now, instead of the `active_eras`.

Reply:
But this method won't be called for the first era - it will only be called once we process a switch block that the current era finalized, by which point we should have all the relevant data of the previous eras (we should have it since initialization, actually).